src/designer_plugin/models.py

```diff
@@ -0,0 +1,186 @@
+"""
+MIT License
+Copyright (c) 2025 Disguise Technologies ltd
+"""
+
+import json
+import traceback
+from dataclasses import dataclass
+from typing import Any, Generic, TypeVar
+
+import typing_extensions
+from pydantic import BaseModel, Field, TypeAdapter, field_validator
+
+###############################################################################
+# Plugin endpoint constants
+D3_PLUGIN_ENDPOINT = "api/session/python/execute"
+D3_PLUGIN_MODULE_REG_ENDPOINT = "api/session/python/registermodule"
+D3_PLUGIN_DEFAULT_PORT = 80
+
+
+###############################################################################
+# Plugin response types
+class PluginStatusDetail(BaseModel):
+    """Detail information for plugin status errors."""
+
+    type_url: str = Field(description="Type URL identifying the error detail.")
+    value: str = Field(description="Error detail value.")
+
+
+class PluginStatus(BaseModel):
+    """Status information from plugin API calls."""
+
+    code: int = Field(description="Return code of plugin API call. 0 if good.")
+    message: str = Field(description="Message from plugin API call.")
+    details: list[PluginStatusDetail] = Field(
+        description="List of detailed error information."
+    )
+
+
+# RetType is the return type of the function
+RetType = typing_extensions.TypeVar("RetType", default=Any)
+RetCastType = TypeVar("RetCastType")
+
+
+class PluginResponse(BaseModel, Generic[RetType]):
+    """Response from a plugin API call."""
+
+    status: PluginStatus = Field(description="Status of plugin API call.")
+    d3Log: str | None = Field(
+        default=None,
+        description="The D3 log field captures the console log for the time the Python script is executing, recording any output generated by running a Python command, including the time taken to execute the Python command. As this captures the Designer console output, it is possible other threads may write to this output during this period causing additional irrelevant output.",
+    )
+    pythonLog: str | None = Field(
+        default=None,
+        description="This output field is the pure Python output, recording any print statements or warnings occurring during the execution of the script. This output will also appear in the d3Log field, however, it may be mixed with other application output.",
```

Comment:
so the pythonLog will not contain other output? The way it's phrased makes it a bit unclear.
```suggestion
        description="This output field is the pure Python output, recording any print statements or warnings that occur during the execution of the script. This output will also appear in the d3Log field. However, in the d3Log field it will be mixed with other Designer log output.",
```